searchlib/src/vespa/searchlib/fef/parameterdescriptions.h

```diff
@@ -37,16 +38,55 @@ struct ParameterCollection {
     };
 };
 
+/*
+ * A set of accepted data types for a parameter.
+ */
+struct ParameterDataTypeSet
+{
+    using DataType = search::index::schema::DataType;
+
+    uint32_t typeMask;
```

Comment:
I prefer not to expose internal state here. Consider making ParamaterDataTypeSet a class instead.